draft-ietf-add-split-horizon-authority.xml

```diff
@@ -179,158 +185,163 @@
+      (see <xref target="vpn"/>).</t>
+      <t>To establish local authority, the network MUST convey one or more
+      "Authorization Claims" to the client.  An "Authorization Claim" is an
+      abstract structure comprising:</t>
+      <ul>
+        <li>An Authentication Domain Name (ADN) of a local encrypted resolver.</li>
+        <li>The DNS name of the authorizing parent zone.</li>
+        <li>A list of subdomains of this parent zone that are claimed by
+            the named local resolver (potentially including the entire parent
+            zone).</li>
+        <li>An NSEC3 Hash Algorithm.</li>
+        <li>A high-entropy salt.</li>
+      </ul>
+      <t>If the local encrypted resolver is identified by name (e.g., DNR), that
+      identifying name MUST be the one used in any corresponding Authorization
+      Claim.  Otherwise (e.g., DDR using IP addresses), the resolver MUST
+      present a validatable certificate containing a subjectAltName that
+      matches the Authorization Claim.</t>
+      <t>To establish its authority, the network MUST provide each Authorization
+      Claim to the parent zone operator.  If the contents are approved, the
+      parent zone operator computes a "Verification Token" according to the
+      following procedure based on <relref section="5" target="RFC5155"/>:</t>
+      <ol>
+        <li>Convert all subdomains into canonical form.</li>
+        <li>Replace the suffix corresponding to the parent zone with a zero
+            byte.</li>
+        <li>Let $X be the concatenation of the resulting pseudo-FQDNs.</li>
+        <li>Let $TOKEN = IH($SALT, $X, 0)</li>
+      </ol>
+      <t>The zone operator then publishes a "Verification Record" with the
+      following structure, following the advice of
+      <xref target="I-D.ietf-dnsop-domain-verification-techniques"/>:</t>
+      <ul>
+        <li>Type = TXT.</li>
+        <li>Owner Name = Concatenation of the ADN, "_splitdns-challenge", and
+            the parent zone name.</li>
+        <li>Contents = "token=base64url($TOKEN)".</li>
+      </ul>
+      <t>By publishing this record, the parent zone authorizes the local
+      encrypted resolver to serve these subdomains authoritatively.</t>
+      <section>
+        <name>Example</name>
+        <t>Consider the following authorization claim:</t>
+        <ul>
+          <li>ADN = "resolver17.parent.example"</li>
+          <li>Parent = "parent.example"</li>
+          <li>Subdomains = "payroll.parent.example",
+              "secret.project.parent.example"</li>
+          <li>Hash Algorithm = SHA-1</li>
+          <li>Salt = "example salt bytes (should be random)"</li>
+        </ul>
+        <t>To approve this claim, the zone operator would publish the following record:</t>
+        <figure><artwork><![CDATA[
+resolver17.parent.example._splitdns-challenge.parent.example. IN TXT \
+"token=kfHudyzZkdQccYVLV0zWsY4ETZUK"
```

Comment:
I think "expiry" key will also have to be included.

Reply:
I'm not crazy about using "_splitdns" in this context but i'm not thikning of anything more palatable

Reply:
Agreed but "expiry" key seems to be mandatory to use. In our case, it will be set to "never".

Reply:
@tireddy2
> Agreed but "expiry" key seems to be mandatory to use.

From [the definition](https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-domain-verification-techniques-01#section-3.1), I don't think that's true.  Some sort of lifetime requirement is mandatory, and "expiry" is the recommended (not required) way to convey it.

Regardless, this draft is not literally an implementation of "Domain Verification".  It is analogous, and reuses the syntax for convenience and consistency, but there is no "provider" in the usual sense.

@moonshiner 
> I'm not crazy about using "_splitdns" in this context but i'm not thikning of anything more palatable

1. Alternatives: "_localdns", "_localview", "_overlay".
2. The "-challenge" suffix is not strictly required here, and could be removed if it seems unhelpful.  (I think it's nice to emphasize the parallelism with Domain Verification.)

Reply:
Thanks @bemasc - I am a terrible bikeshed painter and terrible at naming things.  DNS uses the "view" nomenclature but ignore me for now.  

I do like the work being done here

Reply:
@moonshiner Could also be "_splitview" or "_multiview".  Leaving as "_splitdns" for now.

Reply:
"_splitdns" looks good to me. We should change the title of the draft to "Establishing Local DNS Authority in Validated Split-Horizon Environments"

Reply:
Done